token-group/interface/README.md

```diff
@@ -0,0 +1,138 @@
+
+In addition to demonstrating what a token-group program might look like, it
+also provides some reference examples for using the SPL Type Length Value
+library to manage TLV-encoded data within account data.
+
+For more information on SPL Type Length Value you can reference the library's
+[source code](https://github.com/solana-labs/solana-program-library/tree/master/libraries/type-length-value).
+
+### Motivation
+
+As developers have engineered more creative ways to customize tokens and use
+them to power applications, communities, and more, the reliance on tokens that
+are intrinsically related through on-chain mapping has continued to strengthen.
+
+Token-group provides developers with the minimum necessary interface components
+required to create these relational mappings, allowing for reliable
+composability as well as the freedom to customize these groups of tokens
+however one might please.
+
+By implementing token-group, on-chain programs can build brand-new kinds of
+token groups, which can all overlap with each other and share common tooling.
+
+### Required Instructions
+
+All of the following instructions are listed in greater detail in the source code.
+
+- [`InitializeGroup`](https://github.com/solana-labs/solana-program-library/blob/master/token-group/interface/src/instruction.rs#L22)
+- [`UpdateGroupMaxSize`](https://github.com/solana-labs/solana-program-library/blob/master/token-group/interface/src/instruction.rs#L33)
+- [`UpdateGroupAuthority`](https://github.com/solana-labs/solana-program-library/blob/master/token-group/interface/src/instruction.rs#L42)
+- [`InitializeMember`](https://github.com/solana-labs/solana-program-library/blob/master/token-group/interface/src/instruction.rs#L51)
+
+#### Initialize Group
+
+Initializes a token-group TLV entry in an account for group configurations with
+a provided maximum group size and update authority.
+
+Must provide an SPL token mint and be signed by the mint authority.
+
+#### Update Group Max Size
+
+Updates the maximum size limit of a group.
+
+Must be signed by the update authority.
+
+#### Update Group Authority
+
+Sets or unsets the token-group update authority, which signs any future updates
+to the group configurations.
+
+Must be signed by the update authority.
+
+#### Initialize Member
+
+Initializes a token-group TLV entry in an account for group member
+configurations.
+
+Must provide an SPL token mint for both the group and the group member.
+
+Must be signed by the member mint's mint authority _and_ the group's update
+authority.
+
+### (Optional) State
+
+A program that implements the interface may write the following data fields
+into a type-length-value entry into an account. Note the type discriminants
+for each.
+
+For a group:
+
+```rust
+type OptionalNonZeroPubkey = Pubkey; // if all zeroes, interpreted as `None`
+type PodU32 = [u8; 4];
+type Pubkey = [u8; 32];
+
+/// Type discriminant: [121, 113, 108, 39, 54, 51, 0, 4]
```

Comment:
I didn't check this -- did it come from `"spl_token_group_interface:initialize_token_group"` or `"spl_token_group_interface:group"`?